Docs/(sakuli/sakuli dashboard#60)/how to deploy dashboard (#61)
sakuli/docs: content/enterprise_features/sakuli-dashboard-k8s.md

---
title : "Sakuli Dashboard on Kubernetes"
date :  2019-09-12T14:16:46+02:00
weight : 7
---

# Sakuli Dashboard on Kubernetes

> Once you obtained a Sakuli Enterprise license your docker-user will be granted access to the private Sakuli dashboard image.

Reviewer: ```suggestion
**Attention:** Only Sakuli Enterprise users will be eligible to access the private Sakuli Dashboard Docker images.
```

## Prerequisites

> To setup the Sakuli dashboard on your Kubernetes cluster, it is required to import the image from 
> `taconsol/sakuli-dashboard`. To achieve this, you have to create a docker-registry secret with your `<docker-username>` and
> `<docker-password>` and add it to your service account to authenticate on docker.io during build.

Reviewer: ```suggestion
A Sakuli Dashboard setup on your Kubernetes cluster requires you to import the Sakuli Dashboard image from Docker Hub. In order to do so, please configure a secret storing your `<docker-username>` and `<docker-password>` and add it to your service account for authentication.
```

{{<highlight bash>}}
kubectl create secret docker-registry dockerhub-sakuli-secret \
 --docker-server=docker.io \
 --docker-username=<docker-username> \
 --docker-password=<docker-password> \
 --docker-email=unused
 
kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "dockerhub-sakuli-secret"}]}'
{{</highlight>}}
 

> Now it is possible to import the image from the secured registry.

Reviewer: Maybe we should delete this line.
Since we added the `imagePullSecret` to the default service account there's actually nothing to do here, although it might make you think there is.

> You can start your Sakuli dashboard using two different approaches:
> - Start the Sakuli dashboard via CLI with [kubectl](#kubectl).
> - Start the Sakuli dashboard using a ready to use [template](#dashboard-template).

Reviewer: Aren't we also using `kubectl` to apply the template? 😄 

```suggestion
You can start your Sakuli Dashboard using two different approaches:
- Set up the Sakuli Dashboard manually via [CLI](#k8s-cli).
- Set up the Sakuli Dashboard using a ready to use [template](#k8s-dashboard-template).
```

> ### Starting the Sakuli dashboard with kubectl {#kubectl}

Reviewer: ```suggestion
### Set up the Sakuli Dashboard manually {#k8s-cli}
```

> First you have to create your deployment based on the Sakuli dashboard image you want to use and expose your service.

Reviewer: ```suggestion
As a first step, create a deployment based on the Sakuli Dashboard image and expose the service.
```

> The Sakuli dashboard provides a `latest` tag, which is a tech-preview.
> For a stable version, specify the exact version to ensure consistency.

Reviewer: ```suggestion
Sakuli Dashboard releases are versioned following the [semantic versioning scheme](https://semver.org/).
Images for stable releases are tagged accordingly.
Tech-previews of upcoming stable releases are available via the `latest` tag.

**Attention:** It is highly discouraged to run tech-previews in production as there is no guarantee of stability
```

{{<highlight bash>}}
kubectl create deployment sakuli-dashbaord --image=taconsol/sakuli-dashboard:<IMAGE_TAG>

kubectl expose deployment sakuli-dashboard --type=LoadBalancer --port=8080
{{</highlight>}}

The `--type=LoadBalancer` flag is important to make your service available outside your cluster.

> Now you have to add your Sakuli dashboard configurations such as the dashboard, action, cluster and cronjob configs and your 
> Sakuli license key to the environment of your deployment.

Reviewer: ```suggestion
Now add your Sakuli dashboard configurations such as the dashboard, action, cluster and cronjob configs and your 
Sakuli license key to the environment of your deployment.
```

{{<highlight bash>}}
kubectl set env deployment/sakuli-dashboard --overwrite \
 DASHBOARD_CONFIG="${DASHBOARD_CONFIG}" \
 ACTION_CONFIG="${ACTION_CONFIG}" \
 CLUSTER_CONFIG="${CLUSTER_CONFIG}" \
 CRONJOB_CONFIG="${CRONJOB_CONFIG}" \
 SAKULI_LICENSE_KEY="${SAKULI_LICENSE_KEY}"
{{</highlight>}}

> ### Starting the Sakuli dashboard with a template {#dashboard-template}

Reviewer: ```suggestion
Set up Sakuli Dashboard using a Template {#k8s-dashboard-template}
```

> An XL Sakuli license key is required to start the Sakuli dashboard. You can add your license key with a secret, which is
> then referenced to the deployment in the template.

Reviewer: We do not require the info about an XL license here.

```suggestion
The Sakuli Dashboard Kubernetes template references the Sakuli license key via a secret.
So before applying the template, please add your license key secret first.
```

 
{{<highlight bash>}}
kubectl create secret generic sakuli-license-key \
 --from-literal="SAKULI_LICENSE_KEY=${SAKULI_LICENSE_KEY}"
{{</highlight>}}

> The following config template is ready to use to configure and deploy your Sakuli dashboard.

Reviewer: ```suggestion
The following ready-to-use template will configure a Sakuli Dashboard and deploy it to your cluster
```

> Just copy and save the dashboard template bellow as `dashboard-template.yml`. Furthermore, add your dashboard, action, 

Reviewer: ```suggestion
Just copy and save the dashboard template below as `dashboard-template.yml`. Furthermore, add your dashboard, action, 
```
